<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>757</width>
    <height>652</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MANET SIMULATOR</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="StartButton">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>440</y>
      <width>51</width>
      <height>27</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Start the simulation</string>
    </property>
    <property name="text">
     <string>Start</string>
    </property>
   </widget>
   <widget class="QPushButton" name="StopButton">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>440</y>
      <width>51</width>
      <height>27</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Stop the simulation</string>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
   <widget class="QGraphicsView" name="graphicsView">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>21</y>
      <width>520</width>
      <height>520</height>
     </rect>
    </property>
   </widget>
   <widget class="QToolBox" name="toolBox">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>20</y>
      <width>191</width>
      <height>401</height>
     </rect>
    </property>
    <widget class="QWidget" name="SimulatorPage">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>191</width>
       <height>277</height>
      </rect>
     </property>
     <attribute name="label">
      <string>Simulation Settings</string>
     </attribute>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Area </string>
      </property>
     </widget>
     <widget class="QSpinBox" name="DimensionBox">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>0</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>9999</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Duration (s)</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="TimeBox">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>30</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>999999999</number>
      </property>
      <property name="value">
       <number>60</number>
      </property>
     </widget>
     <widget class="QCheckBox" name="KeepBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>100</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Keep old neighbours</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="BidirectionalBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>130</y>
        <width>121</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Bidirectional</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="AnimationBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>88</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Animation</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QComboBox" name="AttachmentBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>200</y>
        <width>171</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QComboBox" name="LaplacianTypeBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>171</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QComboBox" name="WalkModelBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>171</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="NodePage">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>191</width>
       <height>277</height>
      </rect>
     </property>
     <attribute name="label">
      <string>Mobile Settings</string>
     </attribute>
     <widget class="QSpinBox" name="NodeStartBox">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>50</y>
        <width>53</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>9999</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="NodeDegreeBox">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>20</y>
        <width>53</width>
        <height>25</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Number</string>
      </property>
      <property name="buddy">
       <cstring>NodeStartBox</cstring>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Degree</string>
      </property>
      <property name="buddy">
       <cstring>NodeDegreeBox</cstring>
      </property>
     </widget>
     <widget class="QSpinBox" name="NodeVelocityBox">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>80</y>
        <width>53</width>
        <height>25</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Velocity</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="NodeRadiusBox">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>110</y>
        <width>53</width>
        <height>25</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>120</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Radius</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="HubRadiusBox">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>260</y>
        <width>53</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>99999999</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="HubDegreeBox">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>170</y>
        <width>53</width>
        <height>25</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>180</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Degree</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>200</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Number</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>270</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Radius</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="HubVelocityBox">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>230</y>
        <width>53</width>
        <height>25</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="HubStartBox">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>200</y>
        <width>53</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>9999</number>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>230</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Velocity</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>0</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Mobile Nodes</string>
      </property>
      <property name="buddy">
       <cstring>NodeDegreeBox</cstring>
      </property>
     </widget>
     <widget class="QLabel" name="label_21">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Mobile Hubs</string>
      </property>
      <property name="buddy">
       <cstring>NodeDegreeBox</cstring>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="HubPage">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>191</width>
       <height>277</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>ArrowCursor</cursorShape>
     </property>
     <attribute name="label">
      <string>Street Walk Settings</string>
     </attribute>
     <widget class="QSpinBox" name="GridBox">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>10</y>
        <width>50</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>57</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Grid</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="MessagePage">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>191</width>
       <height>277</height>
      </rect>
     </property>
     <attribute name="label">
      <string>Message Settings</string>
     </attribute>
     <widget class="QSpinBox" name="MsgBERBox">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>60</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>20</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>BER 10^-x</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="MaxDistanceBox">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>190</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>10000</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>200</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Max Distance</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Max Hop</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Max Time (ms)</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="MaxHopBox">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>160</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="MaxTimeBox">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>220</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="decimals">
       <number>0</number>
      </property>
      <property name="maximum">
       <double>100000.000000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="BandwidthBox">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>30</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <double>10000.000000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Bandwidth Mbs</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Length Byte</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="MsgLengthBox">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>0</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>10000</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>121</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Message Deleted</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Send Probability</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="MsgSendProbabilityBox">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>90</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QProgressBar" name="ProgressBar">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>560</y>
      <width>511</width>
      <height>23</height>
     </rect>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QPushButton" name="SnapShotButton">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>440</y>
      <width>71</width>
      <height>27</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Start the simulation</string>
    </property>
    <property name="text">
     <string>Capture</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="ReportBox">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>480</y>
      <width>191</width>
      <height>111</height>
     </rect>
    </property>
    <property name="plainText">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>757</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>HubDegreeBox</tabstop>
  <tabstop>HubStartBox</tabstop>
  <tabstop>NodeDegreeBox</tabstop>
  <tabstop>NodeStartBox</tabstop>
  <tabstop>StartButton</tabstop>
  <tabstop>StopButton</tabstop>
  <tabstop>graphicsView</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>StartButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>startSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>600</x>
     <y>465</y>
    </hint>
    <hint type="destinationlabel">
     <x>531</x>
     <y>438</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>StopButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>stopSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>660</x>
     <y>465</y>
    </hint>
    <hint type="destinationlabel">
     <x>689</x>
     <y>380</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>SnapShotButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>snapShot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>670</x>
     <y>465</y>
    </hint>
    <hint type="destinationlabel">
     <x>675</x>
     <y>436</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>HubDegreeBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setHubDegree(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>695</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>709</x>
     <y>129</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>HubVelocityBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setVelocityHub(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>691</x>
     <y>333</y>
    </hint>
    <hint type="destinationlabel">
     <x>709</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NodeVelocityBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setVelocityNode(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>702</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel">
     <x>506</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NodeDegreeBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setNodeDegree(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>676</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>325</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>NodeStartBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setNodeNumber(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>676</x>
     <y>165</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>325</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>HubStartBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setHubNumber(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>676</x>
     <y>315</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>325</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>startSimulation()</slot>
  <slot>stopSimulation()</slot>
  <slot>setHubNumber(int)</slot>
  <slot>setHubDegree(int)</slot>
  <slot>setEndHubNumber(int)</slot>
  <slot>setNodeDegree(int)</slot>
  <slot>setNodeNumber(int)</slot>
  <slot>setVelocityHub(int)</slot>
  <slot>setVelocityNode(int)</slot>
  <slot>setIncrNode(int)</slot>
  <slot>snapShot()</slot>
 </slots>
</ui>
